go/arrow/csv/reader.go

```diff
@@ -796,6 +885,32 @@ func (r *Reader) parseBinaryType(field array.Builder, str string) {
 	field.(*array.BinaryBuilder).Append(decodedVal)
 }
 
+func (r *Reader) parseLargeBinaryType(field array.Builder, str string) {
+	// specialize the implementation when we know we cannot have nulls
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	decodedVal, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		panic("cannot decode base64 string " + str)
+	}
+	field.(*array.BinaryBuilder).Append(decodedVal)
+}
+
+func (r *Reader) parseFixedSizeBinaryType(field array.Builder, str string) {
+	// specialize the implementation when we know we cannot have nulls
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	decodedVal, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		panic("cannot decode base64 string " + str)
+	}
+	field.(*array.FixedSizeBinaryBuilder).Append(decodedVal)
```

Comment:
`Append` will panic if it is the wrong size, can we check the size first to ensure it is the write length for the fixed size binary and error (or truncate / pad) if it is incorrect?